pkg/pillar/containerd/oci.go

```diff
@@ -320,6 +320,9 @@ func (s *ociSpec) UpdateFromDomain(dom *types.DomainConfig) {
 // OCI runtime spec takes precedence). In addition to that each volume is
 // expected to have rootfs subfolder with a full rootfs filesystem
 func (s *ociSpec) UpdateFromVolume(volume string) error {
+	// keep existing mount points, we do not want to change them here
+	oldMounts := make([]specs.Mount, len(s.Mounts))
+	copy(oldMounts, s.Mounts)
 	if f, err := os.Open(filepath.Join(volume, ociRuntimeSpecFilename)); err == nil {
 		defer f.Close()
 		if err = s.Load(f); err != nil {
```

Comment:
I am slightly confused about how your preservation of `s.Mounts` is expected to work with a usecase when we end up here and load an entire spec from its saved json file. Surely, mounts in the json should be taken literally since we are basically replacing the entire spec with a new copy @giggsoff

Reply:
If I correctly remember, we decided to not process anything about mounts coming from oci itself with https://github.com/lf-edge/eve/pull/1958. So, we have the sequence [here](https://github.com/lf-edge/eve/blob/8c6d4ddecf5fec004d4e188f9abc03644c2746aa/pkg/pillar/hypervisor/containerd.go#L57-L71) which one should give us correct spec in the end.